### PR TITLE
Feat/di get all traversible

### PR DESCRIPTION
**Pull request by bigopon**
# Pull Request

## 📖 Description

In order to support layered templating features, (child syntax override global syntax etc), container should be able to traverse with `.getAll()`, similar to how `.get()` works. This PR adds this.

## ⏭ Next Steps

Continue the work at #819 

cc @xenoterracide

### CODE REVIEW

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://53508-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://53511-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://53528-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://53540-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://55274-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

File: packages/kernel/src/di.ts [hunk 1]
```diff
@@ -433,6 +435,7 @@ function createResolver(getter: (key: any, handler: IContainer, requestor: ICont
   };
 }
 
+
```
Comment (fkleuver) [inline packages/kernel/src/di.ts:438]:
```suggestion
```

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://55384-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Pull request merged**